cmd/juju/commands/common.go

```diff
@@ -172,14 +201,21 @@ func addCharmFromURL(client *api.Client, curl *charm.URL, repo charmrepo.Interfa
 	case "cs":
 		if err := client.AddCharm(curl); err != nil {
 			if !params.IsCodeUnauthorized(err) {
-				return nil, errors.Mask(err)
+				return nil, errors.Trace(err)
 			}
 			m, err := csclient.authorize(curl)
 			if err != nil {
-				return nil, errors.Mask(err)
+				if err1, ok := err.(*errors.Err); ok {
```

Comment:
Given my suggestion above, just:
      if err != nil {
            return termsAgreementError(err)
      }
would be sufficient here.

Or perhaps "maybeTermsAgreementError" might more
appropriate.